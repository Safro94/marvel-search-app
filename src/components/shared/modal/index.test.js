import React from 'react';
import { render, wait, waitForElement, fireEvent } from '@testing-library/react';

import Modal from './';

jest.mock('./style', () => {
  return {
    ModalContainer: 'modalContainer',
    Modal: 'modal',
    H1: 'h1',
    Close: 'close',
    CloseContainer: 'closeContainer',
  };
});

describe('<Modal /> ', () => {
  beforeEach(() => jest.clearAllMocks());
  const title = 'Title';
  const onClose = jest.fn();

  it('It should render OK', async () => {
    // Act
    const { container } = render(
      <Modal title={title} onClose={onClose}>
        <h2>Children</h2>
      </Modal>
    );
    await wait();

    // Assert
    expect(container).toMatchSnapshot();
  });

  it('When the close button is clicked, it should called the function passed by prop', async () => {
    // Act
    const { getByTestId } = render(
      <Modal title={title} onClose={onClose}>
        <h2>Children</h2>
      </Modal>
    );
    const close = await waitForElement(() => getByTestId('close'));
    fireEvent.click(close);

    // Assert
    expect(onClose).toBeCalledTimes(1);
  });
});
