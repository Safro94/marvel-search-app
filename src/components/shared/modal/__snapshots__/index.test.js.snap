// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal />  It should render OK 1`] = `
<div>
  <modalcontainer>
    <modal>
      <closecontainer>
        <close
          data-testid="close"
        />
      </closecontainer>
      <h1>
        Title
      </h1>
      <h2>
        Children
      </h2>
    </modal>
  </modalcontainer>
</div>
`;
